R/recode-with-table.R

```diff
@@ -0,0 +1,1033 @@
+      } else if (interval == "[,)") {
+        return_boolean <-
+          data[[compare_columns]] %in% data[[compare_columns]][which(as.numeric(left_boundary) <= data[[compare_columns]] &
+                                                                       data[[compare_columns]] < as.numeric(right_boundary))]
+      } else if (interval == "(,]") {
+        return_boolean <-
+          data[[compare_columns]] %in% data[[compare_columns]][which(as.numeric(left_boundary) < data[[compare_columns]] &
+                                                                       data[[compare_columns]] <= as.numeric(right_boundary))]
+      } else {
+        stop("Invalid Argument was passed")
+      }
+    }
+
+    return(return_boolean)
+  }
+
+# Parse out variables csv
+update_variable_details_based_on_variable_sheet <-
+  function(variable_sheet, variable_details) {
+    # remove conflicting columns from variable details
+    variable_details <-
+      variable_details[,!(
+        names(variable_details) %in% c(
+          pkg.env$columns.VariableType,
+          pkg.env$columns.VariableLabelShort,
+          pkg.env$columns.VariableLabel,
+          pkg.env$columns.Units
+        )
+      )]
+    # Only keep the needed columns
+    variable_sheet <-
+      variable_sheet[, c(
+        pkg.env$columns.Variables,
+        pkg.env$columns.VariableType,
+        pkg.env$columns.VariableLabelShort,
+        pkg.env$columns.VariableLabel,
+        pkg.env$columns.Units
+      )]
+    # merge the labels and data
+    variable_details <-
+      merge(
+        variable_details,
+        variable_sheet,
+        by.x = pkg.env$columns.Variables,
+        by.y = pkg.env$columns.Variables,
+        all.x = TRUE
+      )
+    # remove variables not present in variable_sheet
+    variable_details <-
+      variable_details[variable_details[[pkg.env$columns.Variables]] %in%
+                         variable_sheet[[pkg.env$columns.Variables]],]
+
+    return(variable_details)
+  }
+
+#' Recode NA formatting
+#'
+#' Recodes the NA depending on the var type
+#'
+#' @param cell_value The value inside the recTo column
+#' @param var_type the toType of a variable
+#'
+#' @return an appropriately coded tagged NA
+recode_variable_NA_formating <- function(cell_value, var_type) {
+  recode_value <- NULL
+  if (grepl("NA", cell_value)) {
+    na_value_list <- strsplit(cell_value, ":")[[1]]
+    if (is_equal(var_type, pkg.env$columns.value.CatType)) {
+      recode_value <- paste("NA(", na_value_list[[3]], ")", sep = "")
+    } else {
+      recode_value <- haven::tagged_na(as.character(na_value_list[[3]]))
+    }
+  } else {
+    if (!is_equal(var_type, pkg.env$columns.value.CatType) &&
+        !is_equal(cell_value, "copy")) {
+      cell_value <- as.numeric(cell_value)
+    }
+    recode_value <- cell_value
+  }
+
+  return(recode_value)
+}
+
+recode_derived_variables <-
+  function(recoded_data,
+           variable_being_processed,
+           variables_to_process,
```

Comment:
Should rename this to `variables_details_rows_to_process` to make it clear that this is not a vector of strings